src/core/src/core_logic/PatchAssessor.py

```diff
@@ -81,12 +88,20 @@ def start_assessment(self):
                 else:
                     error_msg = 'Error retrieving available patches: ' + repr(error)
                     self.composite_logger.log_error(error_msg)
+                    assessment_perf_log = {Constants.LogStrings.TASK: Constants.LogStrings.ASSESSMENT, Constants.LogStrings.PACKAGE_MANAGER: self.package_manager_name,
+                                           Constants.LogStrings.NUMBER_OF_TRIALS: str(number_of_tries), Constants.LogStrings.TASK_STATUS: Constants.LogStrings.FAILED,
+                                           Constants.LogStrings.ERROR_MSG: error_msg}
+                    stopwatch.stop_and_write_telemetry(str(self.assessment_perf_log))
                     self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
                     if Constants.ERROR_ADDED_TO_STATUS not in repr(error):
                         error.args = (error.args, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
                     self.status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
                     raise
 
+        assessment_perf_log = {Constants.LogStrings.TASK: Constants.LogStrings.ASSESSMENT, Constants.LogStrings.PACKAGE_MANAGER: self.package_manager_name,
+                               Constants.LogStrings.NUMBER_OF_TRIALS: str(number_of_tries), Constants.LogStrings.TASK_STATUS: Constants.LogStrings.SUCCEEDED,
+                               Constants.LogStrings.ERROR_MSG: ""}
```

Comment:
This can be extracted out to a function set_assessment_perf_logs(), called on line 101 and 91 with the relevant values.

Reply:
Done.